docs/source/fine-performance-metrics.rst

```diff
@@ -0,0 +1,198 @@
+Fine Performance Metrics
+========================
+.. note::
+
+    This is an experimental feature and may rapidly change without a deprecation cycle.
+
+You may want to investigate where your Dask workload spends the majority of its time;
+not only on which tasks, but also *doing what* while running said tasks.
+*Fine performance metrics* are collected automatically by Dask and attempt to answer
+this question.
+
+In order to observe them, you should simply
+
+#. run your workload end-to-end
+#. open the Dask dashboard (default for a LocalCluster: `<http://localhost:8787>`_)
+#. select ``More...`` -> ``Fine Performance Metrics``
+
+Alternatively, if you're using Jupyter Lab and `dask-labextension
+<https://github.com/dask/dask-labextension>`_, you can just drag the
+``Fine Performance Metrics`` widget onto your Jupyter dashboard.
+
+.. image:: images/fine-performance-metrics/seconds-full.png
+   :alt: Populated Fine Performance Metrics dashboard
+
+The central panel shows, of all the time that the cluster spent doing something, what
+*activities* it spent performing. The most important ones are:
+
+``thread-cpu``
+  CPU time spent by tasks while running on workers
+``thread-noncpu``
+  Difference between wall clock time and CPU time spent by tasks while running on
+  workers. This is typically either I/O time, CPU contention, or GIL contention.
+``idle``
+  Time where a worker had a free thread, but nothing to run on it. This is typically
+  caused by the workload not being able to fully utilize all threads on the cluster,
+  excessive CPU load on the scheduler, or network latency between scheduler and workers.
+  This measure does not include time spent while the whole cluster was completely idle.
+
+The grand total of the time shown should roughly add up to the end-to-end runtime of
+your workload, multiplied by the number of threads on the cluster.
+
+The plot on the right shows network transfer time. Note that most of it should be
+pipelined with task execution, so it may not have an impact. You should worry about this
+only if you have a very large ``idle`` time.
```

Comment:
Do you have an example illustrating a worrisome case?

Reply:
Not at hand. I would need to manually run all tests from coiled/benchmarks that show idle time. Even then, it would be complicated to figure out the cause.
This can be fixed by metering waiting time - but not something we can do now.